allennlp/modules/attention/linear_attention.py

```diff
@@ -0,0 +1,61 @@
+import math
+
+import torch
+from torch.nn import Parameter
+
+from allennlp.nn import util
+from allennlp.nn.activations import Activation
+
+from allennlp.nn.util import masked_softmax
+from overrides import overrides
+
+from allennlp.common.params import Params
+from allennlp.modules.attention.attention import Attention
+
+@Attention.register("linear")
+class LinearAttention(Attention):
+
+    def __init__(self,
+                 tensor_1_dim: int,
+                 tensor_2_dim: int,
+                 combination: str = 'x,y',
+                 activation: Activation = Activation.by_name('linear')(),
+                 normalize: bool = True) -> None:
+        super(LinearAttention, self).__init__()
+        self._normalize = normalize
+        self._combination = combination
+        combined_dim = util.get_combined_dim(combination, [tensor_1_dim, tensor_2_dim])
+        self._weight_vector = Parameter(torch.Tensor(combined_dim))
+        self._bias = Parameter(torch.Tensor(1))
+        self._activation = activation
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        std = math.sqrt(6 / (self._weight_vector.size(0) + 1))
+        self._weight_vector.data.uniform_(-std, std)
+        self._bias.data.fill_(0)
+
+
+    @overrides
+    def forward(self,  # pylint: disable=arguments-differ
+                vector: torch.Tensor,
+                matrix: torch.Tensor,
+                matrix_mask: torch.Tensor = None) -> torch.Tensor:
+        tiled_vector = vector.unsqueeze(1).expand(vector.size()[0],
```

Comment:
@joelgrus @matt-gardner 
I am facing the same issue here too, I don't think this impl will be more efficient than the legacy impl because we can't really avoid the tiling. 
For example: 
In the scenario where the combination is "x,y". 
 we will need to tile x before concatenating  the 2 tensors along the last axis. 
Am I correct in saying that?